src/commands/projects/close.js

```diff
@@ -0,0 +1,141 @@
+ * /projects/:project_id
+ *
+ * There are three 'list' endpoints for projects in the Octokit API.
+ * This command uses one of these list endpoints to get all projects for `account_type`.
+ * Then it can get the ID of the relevant project board, from that list.
+ *
+ * If the `account_type` is 'user', it lists a user's project boards.
+ * If the `account_type` is 'org', it lists a organization's project boards.
+ * If the `account_type` is 'repo', it lists a repository's project boards. This type requres a `repo` argument.
+ *
+ * @see: https://octokit.github.io/rest.js/#api-Projects-listForUser
+ * const result = await octokit.projects.listForUser({username, state, per_page, page})
+ * /users/:username/projects
+ *
+ * @see: https://octokit.github.io/rest.js/#api-Projects-listForOrg
+ * const result = await octokit.projects.listForOrg({org, state, per_page, page})
+ * /orgs/:org/projects
+ *
+ * @see: https://octokit.github.io/rest.js/#api-Projects-listForRepo
+ * const result = await octokit.projects.listForRepo({owner, repo, state, per_page, page})
+ * /repos/:owner/:repo/projects
+ */
+const flow = require('lodash.flow')
+
+const commonYargs = require('../../../lib/common-yargs')
+const printOutput = require('../../../lib/print-output')
+const authenticatedOctokit = require('../../../lib/octokit')
+
+/**
+ * yargs builder function.
+ *
+ * @param {import('yargs').Yargs} yargs - Instance of yargs
+ */
+const builder = yargs => {
+	const baseOptions = flow([
+		commonYargs.withToken,
+		commonYargs.withJson,
+		commonYargs.withOwner, // This is either an organisation or a user
+		commonYargs.withRepo,
+		commonYargs.withNumber,
+		commonYargs.withAccountType,
+	])
+	return baseOptions(yargs)
+}
+
+/**
+ * Get a project based on its number from a list of projects.
+ *
+ * @param {*} number
+ * @param {*} projects
+ */
+const getProject = (number, projects) => projects.data.find(project => {
+	return project.number === number
+})
+
+/**
+ * Update a project board to 'status: closed'.
+ *
+ * @param {object} argv - argv parsed and filtered by yargs
+ * @param {string} argv.token
+ * @param {string} argv.json
+ * @param {string} argv.owner
+ * @param {string} argv.repo
+ * @param {string} argv.number
+ * @param {string} argv.account_type
+ * @throws {Error} - Throws an error if any required properties are invalid
+ */
+const handler = async ({ token, json, owner, repo, number, account_type }) => {
+
+	// Ensure that all required properties have values
+	const requiredProperties = {
+		owner,
+		number,
+		account_type,
+	}
+	if (
+		Object.values(requiredProperties).some(property => !property)
+		|| (account_type === 'repo' && !repo)
+	) {
+		const message = `Please provide all required properties: ${Object.keys(requiredProperties).join(', ')}`
+		if (account_type === 'repo' && !repo) message += `, and 'repo'.`
+		throw new Error(message)
+	}
```

Comment:
Same thing as in `add-pull-request.js`. We should avoid doing validation in the handler.